plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/ExampleMultiParent.java

```diff
@@ -0,0 +1,80 @@
+package org.jenkins.tools.test.hook;
+
+import org.jenkins.tools.test.SCMManagerFactory;
+import org.jenkins.tools.test.model.PluginCompatTesterConfig;
+import org.jenkins.tools.test.model.PomData;
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeCheckout;
+
+import org.apache.maven.scm.ScmFileSet;
+import org.apache.maven.scm.ScmTag;
+import org.apache.maven.scm.command.checkout.CheckOutScmResult;
+import org.apache.maven.scm.manager.ScmManager;
+import org.apache.maven.scm.repository.ScmRepository;
+import hudson.model.UpdateSite.Plugin;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+/**
+ * An example hook for dealing with a multiplugin repository. Since the plugin-compat-tester assumes 
+ * each plugin is in it's own repository, these plugins automatically fail since they are "not found".
```

Comment:
The hook enables you to use whatever configuration you'd like rather than basically having a huge try-catch to figure out how this particular plugin is structured.  You can directly tell the PCT how to checkout the plugin, no matter your configuration.